dspace-api/src/test/java/org/dspace/app/packager/PackagerIT.java

```diff
@@ -0,0 +1,186 @@
+            String idStr = getID();
+            communityService.delete(context, child1);
+            performImportScript(tempFile);
+            Community community = communityService.find(context, UUID.fromString(idStr));
+            assertNotNull(community);
+        } finally {
+            tempFile.delete();
+        }
+    }
+
+    @Test
+    public void packagerUUIDAlreadyExistTest() throws Exception {
+        context.turnOffAuthorisationSystem();
+        createTemplate();
+        createFiles();
+        try {
+            //Item should be overwritten if UUID already Exists
+            performExportScript(article.getHandle(), tempFile);
+            performImportScript(tempFile);
+            Iterator<Item> items = itemService.findByCollection(context, col1);
+            assertEquals(1, Iterators.size(items));
+        } finally {
+            tempFile.delete();
+        }
+    }
+
+    protected void createTemplate() {
+        parentCommunity = CommunityBuilder.createCommunity(context)
+                .withName("Parent Community")
+                .build();
+        child1 = CommunityBuilder.createSubCommunity(context, parentCommunity)
+                .withName("Sub Community")
+                .build();
+        col1 = CollectionBuilder.createCollection(context, child1).withName("Collection 2").build();
+
+        // Create a new Publication (which is an Article)
+        article = ItemBuilder.createItem(context, col1)
+                .withTitle("Article")
+                .withIssueDate("2017-10-17")
+                .withEntityType("Publication")
+                .build();
+    }
+
+    protected void createFiles() throws IOException {
+        tempFile = File.createTempFile("packagerExportTest", ".zip");
+    }
+
+    private String getID() throws IOException, MetadataValidationException {
+        //this method gets the UUID from the mets file thats stored in the attribute element
+        METSManifest manifest = null;
+        ZipFile zip = new ZipFile(tempFile);
+        ZipEntry manifestEntry = zip.getEntry(METSManifest.MANIFEST_FILE);
+        if (manifestEntry != null) {
+            // parse the manifest and sanity-check it.
+            manifest = METSManifest.create(zip.getInputStream(manifestEntry),
+                    false, "AIP");
+        }
+        Element mets = manifest.getMets();
+        String idStr = mets.getAttributeValue("ID");
+        if (idStr.contains("DB-ID-")) {
+            idStr = idStr.substring(idStr.lastIndexOf("DB-ID-") + 6, idStr.length());
+        }
+        return idStr;
+    }
+
+
+    private void performExportScript(String handle, File outputFile) throws Exception {
+        runDSpaceScript("packager", "-d", "-e", "admin@email.com", "-i", handle, "-t",
+                "AIP", outputFile.getPath());
+    }
+
+    private void performImportScript(File outputFile) throws Exception {
+        runDSpaceScript("packager", "-r", "-f", "-u", "-e", "admin@email.com", "-t",
```

Comment:
This looks to always run in "force replace mode" (`-r -f`).  We should have a test which also runs simple in restore more (`-r`).  That test should NOT overwrite an existing object with the same UUID, as "restore" mode means it will throw an error if the object already exists in the system. See docs at https://wiki.lyrasis.org/display/DSDOC7x/AIP+Backup+and+Restore#AIPBackupandRestore-DefaultRestoreMode

Reply:
I created another test to make sure that if the uuid already exists that it will not replace the item when not in force mode and that the item already there isn't merged over. I also fixed an issue with the WorkspaceItemTest because of the new auths the spy wasn't listening for them so thats where the change in that file came from.